packages/simulator/lib/taxiDispatch.js

```diff
@@ -1,23 +1,37 @@
 const { toArray, map, filter, mergeMap, take } = require('rxjs/operators')
 const { plan } = require('./vroom')
 
-const passengerToShipment = (
-  { id, destination, pickup, position, arrivalTime, departureTime },
-  i
-) => ({
-  id: i,
-  description: id,
+const journeyToShipment = (id, journey) => ({
+  id,
+  description: journey.description,
   amount: [1],
   delivery: {
-    id: i,
-    location: [destination.lon, destination.lat],
+    id,
+    location: [journey.destination.lon, journey.destination.lat],
+    time_windows: journey.timeWindow,
   },
   pickup: {
-    id: i,
-    location: [pickup.lon, pickup.lat],
+    id,
+    location: [journey.pickup.lon, journey.pickup.lat],
+    time_windows: journey.timeWindow,
   },
 })
 
+const passengersToShipments = (passengers) => {
+  const jobMap = {}
+  console.log("passengersToShipments")
+  let journeyIndex = 0
+  const shipments = [...passengers].flatMap((passenger, passengerIndex) =>
```

Comment:
Borde det inte vara `from([...passengers]).pipe(flatMap`  - eller hur fungerar detta?

Reply:
I'm going to be honest and say I wasn't sure what to do here. I only know that I wanted to avoid mutating the  `passengers`-instance.

Reply:
This is not `rxjs/operators/flatmap`, it's `Array.prototype.flatMap`, which is the same as `map().flat()`. :) This is a function called inside a `rxjs/operators/map` callback, so it's not part of the stream processing, it's being handled in chunks I guess?!?